app/bundles/CoreBundle/Controller/AbstractFormController.php

```diff
@@ -237,19 +237,17 @@ protected function copyErrorsRecursively(Form $copyFrom, Form $copyTo)
     /**
      * generate $postActionVars with respect to available referer.
      *
-     * @param array $postActionVars
+     * @param array $vars
      *
-     * @return array $postActionVars
+     * @return array
      */
     protected function refererPostActionVars($vars)
     {
         if (empty($this->request->server->get('HTTP_REFERER'))) {
             return $vars;
         }
 
-        $returnUrl                              = !empty($this->request->server->get('HTTP_REFERER'))
-                                                ? $this->request->server->get('HTTP_REFERER')
-                                                : $returnUrl;
+        $returnUrl                              = $this->request->server->get('HTTP_REFERER');
```

Comment:
This change does not test for the value of the request existing before setting the value.

Reply:
Also, $returnUrl isn't defined.

Reply:
Please take another look. This line was faulty to begin with as $returnUrl is never defined, and line 246 already checks for an empty referrer.